calico_mesos/calico_mesos.py

```diff
@@ -259,19 +281,35 @@ def _isolate(hostname, ns_pid, container_id, ipv4_addrs, ipv6_addrs, profiles, l
                                    ip_list=ipv4_addrs)
 
     # Create any profiles in etcd that do not already exist
-    if profiles == []:
-        profiles = ["default"]
+    assigned_profiles = []
     _log.info("Assigning Profiles: %s" % profiles)
+    # First remove any keyword profile names
+    try:
+        profiles.remove("public")
+    except ValueError:
+        pass
+    else:
+        _log.info("Assigning Public Profile")
+        _create_profile_for_public_communication("public")
```

Comment:
Inconsistency here in that for other `_create_profile_...` calls, you check if the profile exists first.